Central unit/feature/sh 168 smart home cli

DyczekPawel opened the pull request:
Refactored IPC layer with socket server architecture and added CLI tool.

* Implemented socket server with TCP and UDS support.
* Created SocketConnection base class with server-side variant.
* Added smarthomectl for system interaction via command line.
* Set up documentation structure with namespace docs and Doxygen without configuration.
* Cleaned up code with const correctness and naming convention fixes.

DyczekPawel committed:
SH-168 Clean up code

 - Remove redundant and unused code fragments
 - Add explicit to TcpConnection constructor
 - Add const where appropriate
 - Fix misspelling in documentation comments
 - Move includes that are needed for config_manager.tpp from header

DyczekPawel committed:
SH-168 Add error handling for failed PID write to lock file

DyczekPawel committed:
SH-168 Changed variable name and namespace alias

Static variable defaultConfigPath is now named sDefaultConfigPath .
Namespace bip alias for boost::asio::ip was changed to bai.

DyczekPawel committed:
SH-168 Rework IPC

 - IPC library now implements socket server
 - Socket server handles UDS and TPC connections
 - IPC library now implements socket connection class and its server version
 - Socket connection handles communication in both TCP and UDS mode
 - Changed YAML config  to implement additional settings for UDS
 - Changed core and main to implement new IPC

DyczekPawel committed:
SH-168 Add basic CLI

CLI can connect to smarthome main process via IPC (TCP or UDS).
Synchronously sends messages from stdin to server and awaits for response
which then is passed to stdout.

DyczekPawel committed:
SH-168 Add namespace documentation, docs folder for doxygen

New file namespaces.h in docs directory contains namespaces documentation.
Create empty Doxyfile with TODO to configure doxygen.

DyczekPawel committed:
SH-168 Remove old TODO

KarasinskiKacper commented on central_unit/src/common/ipc/socket_connection.h:
Wydaje mi się że o ile przy `@param`, `@return` i przy punktach wymienianych od myślników kropka na końcu jest opcjonalna to przy `@note` i `@warning` to na końcu powinna znaleźć się kropka lub wykrzyknik.

Według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI.

KarasinskiKacper commented on central_unit/src/cli/main.cpp:
według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI

KarasinskiKacper commented on central_unit/src/cli/main.h:
według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server_connection.cpp:
według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI

KarasinskiKacper commented on central_unit/src/common/ipc/socket_connection.cpp:
według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server.cpp:
Według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI.

DyczekPawel replied:
W tym plików nie ma const'ów do poprawy, według ustanleń tylko const'y definiowane wewnątrz headerów powinny być tak nazwane.

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server.cpp:
```diff
@@ -0,0 +1,229 @@
+
+    void SocketServer::acceptorHandler(const std::shared_ptr<SocketServerConnection> &connection,
+                                       ba::io_context *ioContext, const bs::error_code &ec,
+                                       const SocketConnection::Type type) {
```
Tutaj drobna uwaga stylistyczna, jak się z nią nie zgadzasz to możesz to zostawić jak chcesz. 

Moim zdaniem jak masz dużo zmiennych do przekazana funkcji, to bardziej czytelne jest formatowanie: 
<img width="205" height="249" alt="image" src="https://github.com/user-attachments/assets/0dcf7d64-595c-4d1c-b09d-9cc9e2b9740c" />
niż:
<img width="407" height="175" alt="image" src="https://github.com/user-attachments/assets/fc3281d5-4d8f-48bb-ba32-bf19727b595d" />

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server.cpp:
```diff
@@ -0,0 +1,229 @@
+        std::lock_guard lock(mActiveConnectionsMutex);
+
+        // TODO consider using another method then iteration from 1
+        for (uint32_t id = 1; id < std::numeric_limits<uint32_t>::max(); ++id) {
```
jak coś zamiast `std::numeric_limits<uint32_t>::max()` można użyć `UINT_MAX`

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server.h:
```diff
@@ -0,0 +1,195 @@
+     *          Accepts incoming connections and manages their lifecycle.
+     *          Supports multiple concurrent connections with unique IDs.
+     *
+     * @note Singleton pattern ensures single server instance
```
Brakuje tu '.'. W pozostałych miejscach przy `@param`, `@post` itp. też, ale to jak pisałem wcześniej:
Wydaje mi się że o ile przy `@param`, `@return` i przy punktach wymienianych od myślników kropka na końcu jest opcjonalna to przy `@note` i `@warning` to na końcu powinna znaleźć się kropka lub wykrzyknik.

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server.h:
Według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI.

DyczekPawel replied:
W tym plików nie ma const'ów do poprawy, według ustanleń tylko const'y definiowane wewnątrz headerów powinny być tak nazwane.

KarasinskiKacper commented on central_unit/src/common/ipc/socket_server_connection.h:
Wydaje mi się że o ile przy @param, @return i przy punktach wymienianych od myślników kropka na końcu jest opcjonalna to przy @note i @warning to na końcu powinna znaleźć się kropka lub wykrzyknik.

Według naszej konwencji na dysku const'y miały być pisane DUŻYMI_LITERAMI.

KarasinskiKacper reviewed (changes requested):
Główne błędy to brak '.' w opisach i nazwy zmiennych "const" pisane małymi literami.

DyczekPawel committed:
SH-168 Fix const naming in headers

Const variables defined in headers now follow project naming convention.
Added missing prefixes.

DyczekPawel committed:
SH-168 Fix documentation comments

DyczekPawel committed:
SH-168 Clean up code

DyczekPawel committed:
SH-168 Add missing code cleanup, fix missing prefix

Add files that were missed in previous commit.
Fixed missing prefix.
